omod-2.0/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/VisitConfigurationController2_0.java

```diff
@@ -0,0 +1,196 @@
+
+		VisitConfiguration visitConfiguration = new VisitConfiguration();
+		visitConfiguration.setEnableVisits(getEnableVisitsValue(administrationService));
+		visitConfiguration
+				.setEncounterVisitsAssignmentHandler(getVisitEncounterHandlerValue(administrationService, encounterService));
+		visitConfiguration.setStartAutoCloseVisitsTask(getAutoCloseVisitsTaskStartedValue(schedulerService));
+		visitConfiguration.setVisitTypesToAutoClose(getVisitTypesToAutoCloseValue(administrationService, visitService));
+
+		return ConversionUtil.convertToRepresentation(visitConfiguration, Representation.FULL);
+	}
+
+	@RequestMapping(method = RequestMethod.POST)
+	@ResponseStatus(HttpStatus.OK)
+	public void updateCurrentConfiguration(@RequestBody VisitConfiguration newConfiguration) {
+		Context.requirePrivilege(PrivilegeConstants.CONFIGURE_VISITS);
+		AdministrationService administrationService = Context.getAdministrationService();
+		VisitService visitService = Context.getVisitService();
+		SchedulerService schedulerService = Context.getSchedulerService();
+
+		// validate
+		if (newConfiguration.getEnableVisits() && StringUtils.isEmpty(newConfiguration.getEncounterVisitsAssignmentHandler())) {
+			throw new IllegalRequestException("Visit encounter handler cannot be empty");
+		}
+
+		updateEnableVisitsValue(administrationService, newConfiguration.getEnableVisits());
+
+		if (newConfiguration.getEnableVisits()) {
+			updateVisitEncounterHandlerValue(administrationService,
+					newConfiguration.getEncounterVisitsAssignmentHandler());
+		}
+		updateGetAutoCloseVisitsTaskStartedValue(schedulerService, newConfiguration.getStartAutoCloseVisitsTask());
+		updateVisitTypesToAutoCloseValue(administrationService, visitService, newConfiguration.getVisitTypesToAutoClose());
+	}
+
+	private Boolean getEnableVisitsValue(AdministrationService administrationService) {
+		String enableVisits = administrationService
+				.getGlobalProperty(OpenmrsConstants.GLOBAL_PROPERTY_ENABLE_VISITS, Boolean.FALSE.toString());
+		return Boolean.parseBoolean(enableVisits);
+	}
+
+	private void updateEnableVisitsValue(AdministrationService administrationService, Boolean value) {
+		GlobalProperty enableVisitsGp = new GlobalProperty(OpenmrsConstants.GLOBAL_PROPERTY_ENABLE_VISITS,
+				Boolean.toString(value));
+		administrationService.saveGlobalProperty(enableVisitsGp);
+	}
+
+	private String getVisitEncounterHandlerValue(AdministrationService administrationService,
+			EncounterService encounterService) {
+		String visitEncounterHandler = administrationService.getGlobalProperty(OpenmrsConstants.GP_VISIT_ASSIGNMENT_HANDLER);
+		for (EncounterVisitHandler visitHandler : getEncounterVisitHandlers(encounterService)) {
+			if (visitHandler.getClass().getName().equals(visitEncounterHandler)) {
+				return visitHandler.getClass().getName();
+			}
+		}
+		return null;
+	}
+
+	private Collection<EncounterVisitHandler> getEncounterVisitHandlers(EncounterService encounterService) {
```

Comment:
Do we need this method?

Reply:
https://github.com/openmrs/openmrs-module-webservices.rest/pull/489/commits/a7778cf3d0d939613e2290733a39921237c4a57e